test/phoenix_live_view/integrations/expensive_runtime_checks_test.exs

```diff
@@ -0,0 +1,105 @@
+defmodule Phoenix.LiveViewTest.ExpensiveRuntimeChecksTest do
+  # this is intentionally async: false as we change the application
+  # environment and recompile files!
+  use ExUnit.Case, async: false
+
+  import ExUnit.CaptureIO
+
+  import Phoenix.ConnTest
+
+  import Phoenix.LiveViewTest
+  alias Phoenix.LiveViewTest.Endpoint
+
+  @endpoint Endpoint
+
+  setup_all do
+    # silence redefining module warnings
+    Application.put_env(:phoenix_live_view, :enable_expensive_runtime_checks, true)
```

Comment:
Maybe we could enable it in `config/test.exs`? It is fine to always be on during our own test suite.